File: client-api/src/main/java/org/ostelco/topup/api/resources/ResourceHelpers.java
package org.ostelco.topup.api.resources;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Common 'helper' functions for resources.
 *
 */
public abstract class ResourceHelpers {
    private static final Logger LOG = LoggerFactory.getLogger(ResourceHelpers.class);

    private final ObjectMapper MAPPER = new ObjectMapper();

    protected String asJson(final Object object) {
        try {
            return MAPPER.writeValueAsString(object);
        } catch (JsonProcessingException e) {
            LOG.error("Error in json response {}", e);
        }
        return "";
    }
}

File: client-api/src/main/kotlin/org/ostelco/prime/client/api/resources/ResourceHelpers.kt
package org.ostelco.prime.client.api.resources

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import org.ostelco.prime.logger

/**
 * Common 'helper' functions for resources.
 *
 */
val objectMapper = ObjectMapper()

fun <R : Any> R.asJson(`object`: Any): String {
    try {
        return objectMapper.writeValueAsString(`object`)
    } catch (e: JsonProcessingException) {
        val logger by logger()
        logger.error("Error in json response {}", e)
    }
    return ""
}